src/commands/api/update.js

```diff
@@ -70,6 +51,40 @@ class UpdateAPICommand extends BaseCommand {
     if (flags.publish) await publish.run([apiPathWithVersion])
     if (flags.setdefault) await setDefault.run([apiPathWithVersion])
   }
+
+  async handleUpdate(owner, name, apiVersion, flags) {
+    await this.executeHttp({
```

Comment:
The main part of the change is here. Previously we were calling the API to get the particular API version (`/apis/{owner}/{api}/{version}`) to verify that a user is actually updating an existing API. Now we call the API to get the current visibility setting for the API (`/apis/{owner}/{api}/{version}/settings/private`). If the user did not provide the `visibility` flag, we just post back the existing visibility setting, so the value is not changed unexpectedly.

We have to do it this way because the post API (`/apis/{owner}/{api}`) has a default value for visibility (`isPrivate`).

I made some further code simplifications in this file as the code was getting difficult to understand even before my change, after my change it got worse.